src/serializers/infer.rs

```diff
@@ -672,3 +642,71 @@ pub(crate) fn infer_json_key_known<'py>(ob_type: ObType, key: &'py PyAny, extra:
         }
     }
 }
+
+fn serialize_pairs_python<'py>(
+    py: Python,
+    pairs_iter: impl Iterator<Item = PyResult<(&'py PyAny, &'py PyAny)>>,
+    include: Option<&PyAny>,
+    exclude: Option<&PyAny>,
+    extra: &Extra,
+) -> PyResult<PyObject> {
+    let new_dict = PyDict::new(py);
+    let filter = AnyFilter::new();
+
+    for result in pairs_iter {
+        let (k, v) = result?;
+        let op_next = filter.key_filter(k, include, exclude)?;
+        if let Some((next_include, next_exclude)) = op_next {
+            let v = infer_to_python(v, next_include, next_exclude, extra)?;
+            new_dict.set_item(k, v)?;
+        }
+    }
+    Ok(new_dict.into_py(py))
+}
+
+fn serialize_pairs_python_mode_json<'py>(
+    py: Python,
+    pairs_iter: impl Iterator<Item = PyResult<(&'py PyAny, &'py PyAny)>>,
+    include: Option<&PyAny>,
+    exclude: Option<&PyAny>,
+    extra: &Extra,
+) -> PyResult<PyObject> {
+    let new_dict = PyDict::new(py);
+    let filter = AnyFilter::new();
+
+    for result in pairs_iter {
+        let (k, v) = result?;
+        let op_next = filter.key_filter(k, include, exclude)?;
+        if let Some((next_include, next_exclude)) = op_next {
+            let k_str = infer_json_key(k, extra)?;
+            let k = PyString::new(py, &k_str);
```

Comment:
annoyingly this method is only different here, I guess we could have some trait for this logic, then two implementations, but it seemed like overkill.